teps/0044-composing-tasks-with-tasks.md

```diff
@@ -0,0 +1,137 @@
+---
+status: proposed
+title: Composing Tasks with Tasks
+creation-date: '2021-01-22'
+last-updated: '2021-01-22'
+authors:
+- '@bobcatfish'
+---
+
+# TEP-0044: Composing Tasks with Tasks
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [PipelineResources](#pipelineresources)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [References (optional)](#references-optional)
+
+## Summary
+
+This TEP describes a gap in composability that isn't currently fully solved by Tasks or Pipelines
+(tho PipelineResources do solve some of it!). 
+
+Ideally by addressing the issues described here we'd make it possible to create more efficient
+Pipelines and make it easier to use a Task easily via a TaskRun, and we'd address some of
+[the problems PipelineResources are trying to solve](https://docs.google.com/document/d/1KpVyWi-etX00J3hIz_9HlbaNNEyuzP6S986Wjhl3ZnA/edit#)
+making it more clear whether or not they are an abstraction we want to keep.
+
+[#1673: Pipeline Resources Redesign](https://github.com/tektoncd/pipeline/issues/1673)
+
+## Motivation
+
+Currently the only way to combine Tasks together is in a Pipeline. If you combine
+Tasks in a Pipeline and they need to share data (beyond a simple
+[result](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#using-results)),
+you'll need to provision a PVC or do some other similar, cloud specific thing,
+to [make a volume available](https://github.com/tektoncd/pipeline/blob/master/docs/workspaces.md#specifying-volumesources-in-workspaces)
+that can be shared between them.
+
+Some problems with this:
+* PVCs add additional overhead, both in speed and in management (somewhat
```

Comment:
A separate but similar problem (and possibly an additional use-case): an organization may have already decided on cloud buckets as their means of data sharing. Then Tekton comes along "requiring" k8s volumes and so the org is unable or unwilling to adopt Tekton.

I've seen this reported in Slack as a downside of Workspaces though unfortunately don't have a Github issue / user report to point at.

Composing a bucket-download Task with a build Task with a bucket-upload Task would go some way to solving those users' needs I think.

Reply:
re: buckets -- that's interesting. As an operator I can confirm that allocating the sort of short-lived PVCs typically needed by a pipeline is a real pain. A Task that uploads/downloads to a bucket as a pre/post Task might be useful to eliminate the need for a Pipeline PVC.